Foundation/FileManager.swift

```diff
@@ -992,15 +1003,26 @@ open class FileManager : NSObject {
      */
     open func fileSystemRepresentation(withPath path: String) -> UnsafePointer<Int8> {
         precondition(path != "", "Empty path argument")
+#if os(Windows)
+        return try! _fileSystemRepresentation(withPath: path) {
+            String(decodingCString: $0, as: UTF16.self).withCString() {
+                let sz = strnlen($0, Int(MAX_PATH))
+                let buf = UnsafeMutablePointer<Int8>.allocate(capacity: sz + 1)
+                buf.initialize(from: $0, count: sz + 1)
+                return UnsafePointer(buf)
+            }
+        }
+#else
         return try! __fileSystemRepresentation(withPath: path)
+#endif
     }
 
-    internal func __fileSystemRepresentation(withPath path: String) throws -> UnsafePointer<Int8> {
+    internal func __fileSystemRepresentation(withPath path: String) throws -> UnsafePointer<foundation_char_t> {
         let len = CFStringGetMaximumSizeOfFileSystemRepresentation(path._cfObject)
         if len != kCFNotFound {
-            let buf = UnsafeMutablePointer<Int8>.allocate(capacity: len)
+            let buf = UnsafeMutablePointer<foundation_char_t>.allocate(capacity: len)
             buf.initialize(repeating: 0, count: len)
-            if path._nsObject.getFileSystemRepresentation(buf, maxLength: len) {
+            if path._nsObject._getFileSystemRepresentation(buf, maxLength: len) {
```

Comment:
It looks like there is only a version of `_getFileSystemRepresentation` for Windows but its also used by non-windows platforms or am I missing something?